papergrid/src/lib.rs

```diff
@@ -941,41 +957,39 @@ fn build_row_internal_line(
 ) -> fmt::Result {
     let top_indent = top_indent(cell, style, height);
     if top_indent > line_index {
-        return empty_line(f, width);
+        return repeat_char(f, style.padding.top_char, width);
     }
 
     let cell_line_index = line_index - top_indent;
     let cell_has_this_line = cell.len() > cell_line_index;
     // happen when other cells have bigger height
     if !cell_has_this_line {
-        return empty_line(f, width);
+        return repeat_char(f, style.padding.bottom_char, width);
     }
 
     let line_text = cell[cell_line_index];
     line(
         f,
         line_text,
         width,
-        style.indent.left,
-        style.indent.right,
+        style.padding.left,
+        style.padding.left_char,
+        style.padding.right,
+        style.padding.right_char,
         style.alignment_h,
     )
 }
 
 fn top_indent(cell: &[&str], style: &Style, height: usize) -> usize {
-    let height = height - style.indent.top;
-    let content_height = cell_height(cell, style) - style.indent.top - style.indent.bottom;
+    let height = height - style.padding.top;
+    let content_height = cell_height(cell, style) - style.padding.top - style.padding.bottom;
     let indent = style.alignment_v.top_ident(height, content_height);
-    indent + style.indent.top
-}
-
-fn empty_line(f: &mut std::fmt::Formatter<'_>, n: usize) -> fmt::Result {
-    write!(f, "{:1$}", "", n)
+    indent + style.padding.top
 }
 
 fn repeat_char(f: &mut std::fmt::Formatter<'_>, c: char, n: usize) -> fmt::Result {
     if n > 0 {
-        write!(f, "{:1$}", c, n)
+        write!(f, "{:1$}", c.to_string().repeat(n), n)
```

Comment:
I think that doing write in a loop will be better, because of less allocations?
I might be wrong though.
What do you think?

```rust
for _ in 0 .. n {
  write!(f, "{}", c)?;
}

Ok(())
```

Reply:
I try to find the proper way to profile function in Rust (likeGo or Java) and left it as you offered.